src/config.js

```diff
@@ -71,6 +74,19 @@ class Config {
         }
       }
     }
+
+    this.recordedValueMaxLength = coalesce(
+      parseInt(options.recordedValueMaxLength, 10),
+      parseInt(platform.env('SIGNALFX_RECORDED_VALUE_MAX_LENGTH'), 10),
+      DEFAULT_RECORDED_VALUE_MAX_LENGTH
+    )
+    if (this.recordedValueMaxLength === -1) {
+      this.recordedValueMaxLength = Number.MAX_SAFE_INTEGER
```

Comment:
This is fine but other implementations just store `-1` and then the truncate function bails out if the value is <0